tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -146,8 +147,10 @@ def preprocess_train(self, X_train, y_train, X_val, y_val, **kwargs):
         return data
 
     def fill_missing(self, df):
-        df = df[self.cat_columns + self.cont_columns]
+        df = df[self.cat_columns + self.cont_columns].copy()
         for c in self.cat_columns + self.cont_columns:
+            if c in self.cat_columns:
+                df[c] = df[c].cat.add_categories(MISSING)
```

Comment:
What happens if `MISSING` is already is a category?

Reply:
We need to pick something: this category is supposed to be something not in the training set by design.

Reply:
The comment wasn't about changing the missing value, it was to ensure that we didn't crash if it was present.